### PR TITLE
Adding Button Component

### PR DESCRIPTION
**Pull request by sallamTanna**

### CODE REVIEW

File: client/src/components/Button/index.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import {
+  faArrowLeft,
+  faArrowRight,
+  faCheckCircle
+} from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+
+export default function Button(props) {
+  return (
+        <button className={classes.button} style={props.style} onClick={props.onClick}>
```
Comment (AbdallahAmmar96) [inline client/src/components/Button/index.js]: Use destruction for the props

File: client/src/components/Button/index.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+          {props.leftIcon? <FontAwesomeIcon icon={faArrowLeft}/>: null}
+            <span>{props.children}</span>
```
Comment (AbdallahAmmar96) [inline client/src/components/Button/index.js]: Add className for the span and use it in style 
don't use select element by tags

File: client/src/components/Button/index.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import classes from './style.css'
```
Comment (AbdallahAmmar96) [inline client/src/components/Button/index.js]: add a new line after the import the React

File: client/src/components/Button/index.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import {
+  faArrowLeft,
+  faArrowRight,
+  faCheckCircle
+} from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
```
Comment (AbdallahAmmar96) [inline client/src/components/Button/index.js]: from line 3 to line 8 should be at the top of import the style with a line breaker

File: client/src/components/Button/index.js (hunk 1 above)
Comment (sallamTanna) [reply client/src/components/Button/index.js]: Done

File: client/src/components/Button/index.js (hunk 3 above)
Comment (sallamTanna) [reply client/src/components/Button/index.js]: Done

File: client/src/components/Button/style.css [hunk 5]
```diff
@@ -0,0 +1,18 @@
+.button {
+  text-transform: uppercase;
+  border: none;
+  border-radius: 5px;
+  color: white;
+  letter-spacing: 1.5px;
+  font-weight: 200;
```
Comment (AbdallahAmmar96) [inline client/src/components/Button/style.css]: font-weight is light you can put it 600

File: client/src/components/Button/style.css [hunk 6]
```diff
@@ -0,0 +1,18 @@
+.button {
```
Comment (AbdallahAmmar96) [inline client/src/components/Button/style.css]: add cursor for the button with a value pointer

Comment (sallamTanna) [reply client/src/components/Button/style.css]: Done

File: client/src/components/Button/style.css (hunk 5 above)
Comment (sallamTanna) [reply client/src/components/Button/style.css]: Done

File: client/src/components/Button/index.js (hunk 3 above)
Comment (AbdallahAmmar96) [inline client/src/components/Button/index.js]: you can import the style like this 
```js
import './style.css
```

Comment (sallamTanna) [reply client/src/components/Button/index.js]: Done

File: client/src/components/Button/index.js (hunk 1 above)
Comment (AbdallahAmmar96) [inline client/src/components/Button/index.js]: you can delete the style attribute, no need for it
just add a class name  for the button 
on is custom and the other will depend on where the button used

File: client/src/components/Button/index.js (hunk 4 above)
Comment (sallamTanna) [reply client/src/components/Button/index.js]: Done

File: client/src/components/Button/index.js (hunk 2 above)
Comment (sallamTanna) [reply client/src/components/Button/index.js]: Done

File: client/src/components/Button/index.js (hunk 1 above)
Comment (sallamTanna) [reply client/src/components/Button/index.js]: Done

File: client/src/components/Button/index.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
```
Comment (AbdallahAmmar96) [inline client/src/components/Button/index.js]: Either you put a semicolon or you don't
follow one pattern and use it in all files 
I prefer you don't put a  semicolon in react

File: client/src/components/Button/index.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+  faArrowLeft,
+  faArrowRight,
+  faCheckCircle
+} from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+
+import './style.css'
+
+export default function Button(props) {
+  const { label, style, done, rightIcon, leftIcon, onClick } = props;
```
Comment (AbdallahAmmar96) [inline client/src/components/Button/index.js]: as you named rightIcon and leftIcon make the done => doneIcon

File: client/src/components/Button/index.js [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+  return (
+        <button className={`button ${props.className}`} onClick={onClick}>
+          {leftIcon? <FontAwesomeIcon icon={faArrowLeft} className=''/>: null}
+            <span className='button-title'>{label}</span>
+          {done? <FontAwesomeIcon icon={faCheckCircle} className='' />: null}
+          {rightIcon? <FontAwesomeIcon icon={faArrowRight} className=''/>: null}
+        </button>
```
Comment (AbdallahAmmar96) [inline client/src/components/Button/index.js]: from line 16 to 13 you put an empty class name!! why

File: client/src/components/Button/index.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+import './style.css'
+
+export default function Button(props) {
+  const { label, style, done, rightIcon, leftIcon, onClick } = props;
+  return (
+        <button className={`button ${props.className}`} onClick={onClick}>
+          {leftIcon? <FontAwesomeIcon icon={faArrowLeft} className=''/>: null}
+            <span className='button-title'>{label}</span>
+          {done? <FontAwesomeIcon icon={faCheckCircle} className='' />: null}
+          {rightIcon? <FontAwesomeIcon icon={faArrowRight} className=''/>: null}
```
Comment (AbdallahAmmar96) [inline client/src/components/Button/index.js]: add a space before closing the tag and space after it 
and space before the question mark

File: client/src/components/Button/index.js [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+
+import './style.css'
+
+export default function Button(props) {
+  const { label, style, done, rightIcon, leftIcon, onClick } = props;
+  return (
+        <button className={`button ${props.className}`} onClick={onClick}>
+          {leftIcon? <FontAwesomeIcon icon={faArrowLeft} className=''/>: null}
+            <span className='button-title'>{label}</span>
```
Comment (AbdallahAmmar96) [inline client/src/components/Button/index.js]: indentation !! 
why is the span looks like it's nested in but it's not?

File: client/src/components/Button/index.js (hunk 7 above)
Comment (sallamTanna) [reply client/src/components/Button/index.js]: Done

File: client/src/components/Button/index.js (hunk 8 above)
Comment (sallamTanna) [reply client/src/components/Button/index.js]: Done

File: client/src/components/Button/index.js (hunk 9 above)
Comment (sallamTanna) [reply client/src/components/Button/index.js]: By mistake, i removed them

File: client/src/components/Button/index.js (hunk 10 above)
Comment (sallamTanna) [reply client/src/components/Button/index.js]: Done

File: client/src/components/Button/index.js (hunk 11 above)
Comment (sallamTanna) [reply client/src/components/Button/index.js]: Done

**Pull request merged**